### PR TITLE
test(targeting): add unit tests for different flag condition types

### PR DESCRIPTION
**Pull request by kwalker3690**
### Summary

Add unit tests to track different types of supported conditions

### Checklist

* [x] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-TypeScript/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:  no

### CODE REVIEW

File: packages/targeting/test/flag-config-data/multiple-events.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+          {
+            selector: ['context', 'event_types'],
+            op: 'set contains',
+            values: ['Add to Cart', 'Sign In'],
+          },
+        ],
+      ],
+    },
+    {
+      variant: 'off',
```
Comment (jackson-amplitude) [inline packages/targeting/test/flag-config-data/multiple-events.ts:38]: it doesn't look like this is added to all segments. is this only for specific segment types?

File: packages/targeting/test/targeting.test.ts [hunk 2]
```diff
@@ -71,8 +75,103 @@ describe('targeting', () => {
           session_id: 123,
           event_types: ['Add to Cart', 'Purchase'],
         },
-        [flagConfig],
+        [flagCatchAll],
       );
     });
   });
+
+  describe('condition tests', () => {
```
Comment (lewgordon-amplitude) [inline packages/targeting/test/targeting.test.ts:83]: It'd be nice to create a fixture here instead. Something like I did [here](https://github.com/amplitude/Amplitude-TypeScript/blob/584e1a2aefaea14d344dcb607cc77ec518c9567c/packages/session-replay-browser/test/hooks/click.test.ts#L153). That way we can keep most of the fields that are common and we don't care too much about for a particular test the same. It makes it easier to write smaller tests more quickly. Let me know what you think!

**Review by lewgordon-amplitude — APPROVED** [review]
LGTM!

**Review by jackson-amplitude — APPROVED** [review]
lgtm! Thanks for adding these to make sure we are aligned!

File: packages/targeting/test/targeting.test.ts (hunk 2 above)
Comment (kwalker3690) [reply packages/targeting/test/targeting.test.ts:83]: For sure, I definitely like that approach too, although in some cases I like to see the entirety of a config in a test data file (that was helpful to me when understanding the entirety of the flag config in the first place). I'm going to leave these as is for now but happy to iterate in that direction in a future pr

**Pull request merged**